### ISSUE TITLE
MAX32670EVKIT RevB debugging odd behavior

### ISSUE THREAD

**Issue by fneirab** [MaximIntegratedTechSupport/VSCode-Maxim#6]
![image](https://user-images.githubusercontent.com/23093027/159335845-3cb2a1b2-1c54-4f4e-834c-869953f3b4b6.png)
![image](https://user-images.githubusercontent.com/23093027/159335912-2f37939a-30ab-4a62-8cd2-12f6149a0b21.png)
VSCode-Maxim v1.3.0 release files. 

When debugging the "hello_world" example the debug console produces the following report:

`openocd: Open On-Chip Debugger 0.11.0+dev-g4f85f7914 (2021-12-16-13:53)
openocd: Licensed under GNU GPL v2
openocd: For bug reports, read
openocd: 	[http://openocd.org/doc/doxygen/bugs.html]()
openocd: DEPRECATED! use 'adapter driver' not 'interface'
openocd: DEPRECATED! use 'adapter srst delay' not 'adapter_nsrst_delay'
openocd: DEPRECATED! use 'adapter srst pulse_width' not 'adapter_nsrst_assert_width'
openocd: Info : CMSIS-DAP: SWD  supported
openocd: Info : CMSIS-DAP: FW Version = 1.0
openocd: Info : CMSIS-DAP: Interface Initialised (SWD)
openocd: Info : SWCLK/TCK = 0 SWDIO/TMS = 0 TDI = 0 TDO = 0 nTRST = 0 nRESET = 0
openocd: Info : CMSIS-DAP: Interface ready
openocd: Info : clock speed 2000 kHz
openocd: Info : SWD DPIDR 0x2ba01477
openocd: Info : max32xxx.cpu: Cortex-M4 r0p1 processor detected
openocd: Info : max32xxx.cpu: target has 6 breakpoints, 4 watchpoints
openocd: Info : max32xxx.cpu: external reset detected
openocd: Info : starting gdb server for max32xxx.cpu on 3333
openocd: Info : Listening on port 3333 for gdb connections
openocd: Warn : Only resetting the Cortex-M core, use a reset-init event handler to reset any peripherals or configure hardware srst support.
openocd: target halted due to debug-request, current mode: Thread 
openocd: xPSR: 0x01000000 pc: 0x000001b4 msp: 0x20004000
openocd: Info : Listening on port 6666 for tcl connections
openocd: Info : Listening on port 4444 for telnet connections
=thread-group-added,id="i1"
GNU gdb (GNU Tools for Arm Embedded Processors 9-2019-q4-major) 8.3.0.20190709-git
Copyright (C) 2019 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <[http://gnu.org/licenses/gpl.html>]()
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "--host=i686-w64-mingw32 --target=arm-none-eabi".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<[http://www.gnu.org/software/gdb/bugs/>]().
Find the GDB manual and other documentation resources online at:
    <[http://www.gnu.org/software/gdb/documentation/>]().

For help, type "help".
Type "apropos word" to search for commands related to "word".
openocd: Info : accepting 'gdb' connection on tcp/3333
openocd: Warn : Prefer GDB command "target extended-remote :3333" instead of "target remote :3333"
openocd: Warn : Only resetting the Cortex-M core, use a reset-init event handler to reset any peripherals or configure hardware srst support.
openocd: target halted due to debug-request, current mode: Thread 
openocd: xPSR: 0x01000000 pc: 0x000001b4 msp: 0x20004000
=cmd-param-changed,param="pagination",value="off"
openocd: Info : SWD DPIDR 0x2ba01477
openocd: Error: Failed to read memory at 0xfffff000
openocd: Info : SWD DPIDR 0x2ba01477
openocd: Error: Failed to read memory at 0xfffff000
openocd: Info : SWD DPIDR 0x2ba01477
openocd: Error: Failed to read memory at 0xfffff000
openocd: Info : SWD DPIDR 0x2ba01477
openocd: Error: Failed to read memory at 0xfffff000
Note: automatically using hardware breakpoints for read-only addresses.

Breakpoint 1, main () at main.c:62
62	    printf("Hello World!\n");
Execute debugger commands using "-exec <command>", for example "-exec info registers" will list registers in use (when GDB is the debugger)`

The debug console looks stops at an unspecified breakpoint, but never gets to the first break point and continues the debugging with the second break point working as expected.  
![image](https://user-images.githubusercontent.com/23093027/159336560-4e1a7d4d-8a18-45f2-a667-292e0a998e3d.png)
![image](https://user-images.githubusercontent.com/23093027/159336640-d3b60613-eb6a-4319-b1fa-fda260db7a66.png)

Similar behavior is observed on other examples of the MAX32670EVKIT, like the I2C example. However, in this example when I press the continue button, I get am Unknown break point exception. 
![image](https://user-images.githubusercontent.com/23093027/159338550-21349802-1759-48f8-b5cd-05e22a943293.png)
![image](https://user-images.githubusercontent.com/23093027/159338230-04cae921-946f-4c68-97e0-29ceebc138e9.png)

The terminal output does not match the I2C example:
![image](https://user-images.githubusercontent.com/23093027/159338906-bb6f02d4-e1aa-412d-992d-bafbf79121ff.png)

During compile I do not get any error messages and a I2C.elf is generated. 

 I checked the behavior of of the "Hello_World" example with the MAX32660EVSYS and did not see the same behavior. The test with the MAX32660 was done with the latest release files.

**Comment by Jake-Carter** [comment]
Hi Felipe, this could be the result of a mismatch between the flashed program and the debugger symbols.

With v1.3.0 flashing no longer happens automatically when you launch the debugger.  See [this](https://github.com/MaximIntegratedTechSupport/VSCode-Maxim#launching-the-debugger) section of the readme for more details.

I made this change so that you can quickly restart the program with the green restart arrow.  On the older versions, restarting would do a full re-build and re-flash every time, which made restarting very slow.

So basically just make sure you've got the program manually flashed before you debug.  Then, the debug symbols will match the actual instructions addresses on the micro and I would expect to see this strange behavior go away.

![image](https://user-images.githubusercontent.com/38844790/159363112-06921f3d-377e-474a-bc09-34485d9a2d0c.png)

**Comment by fneirab** [comment]
Hi @Jake-Carter, I tested with the v1.4.1 and I still see the same behavior with the Hello World for the MAX32670EVKIT
![image](https://user-images.githubusercontent.com/23093027/166510593-73671bee-b1e4-4b34-a2ce-ea10f989b478.png) It always start the debugging at the printf statement.

**Comment by Jake-Carter** [comment]
Hi @fneirab looks like everything is working normally.  This is expected behavior.  The debugger will break on entry into main, and the "printf" is the first function it can break on.  It doesn't stop on `int count = 0;` because I believe the compiler will optimize the variable declaration and move it around.  Your debugger should be working normally from here!
